vermouth/processors/do_links.py

```diff
@@ -98,7 +98,7 @@ def _interpret_order(order):
     return order_type, order_value
 
 
-def match_order(order1, resid1, order2, resid2):
+def match_order(order1, resid1, chain1, order2, resid2, chain2):
```

Comment:
Currently you separate chain1, resid1, chain2, resid2 everywhere. Would it make sense to consolidate them into a `residue_identifier`? Could be a tuple (or dict). Does bring a few headaches for dealing with orders +-<>, since that implies you need to be able to subtract and compare them. Or, for those cases, you pull out the resid.
Doesn't make much difference now, but does make adding more criteria later on easier.

Reply:
Yes, this is a hack, and it would be better to have a residue id class that supports comparisons; it wouldn't need a chain ID then, because you can query the corresponding molecule graphs if so needed. That requires a few more lines though, and thus more time than we have now.

Reply:
A `Residue` class would indeed fix a few things. But I completely it's out of scope for this PR. Could you open an issue?